pkg/opt/config.go

```diff
@@ -94,6 +99,25 @@ func (c *Config) VaultClient() (*vaultapi.Client, error) {
 	return nil, nil
 }
 
+func (c *Config) WithMetrics() (*metric.Meter, error) {
```

Comment:
I'd probably just call this method `Metrics()`. Personally, I only use `With` as a prefix for options-passing and methods that take callbacks (and I doubt I've ever verbalized that before) -- not sure if there's a different set of semantics you have, but this might be a nice thing to talk about in the relay-engineering repository at some point.

Reply:
Yeah, that's the convention I used for the other clients. Will update.